pulsar-client/src/main/java/org/apache/pulsar/client/impl/schema/KeyValueSchema.java

```diff
@@ -98,17 +134,29 @@ private KeyValueSchema(Schema<K> keySchema,
         Gson valueSchemaGson = new Gson();
         properties.put("value.schema.properties", valueSchemaGson.toJson(valueSchema.getSchemaInfo().getProperties()));
 
+        if (keyValueEncodingType != null && keyValueEncodingType.equals(KeyValueEncodingType.SEPARATED)) {
```

Comment:
> If you don't allow passing in a null value, you don't need if-else here. your logic is much simpler as below:
> 
> ```
> this.keyValueEncodingType = this.keyValueEncodingType = KeyValueEncodingType.SEPARATED;;
> properties.put("kv.encoding.type", String.valueOf(keyValueEncodingType));
> ```

Is that what you mean? @sijie 
```
this.keyValueEncodingType = (keyValueEncodingType == null || keyValueEncodingType.equals(KeyValueEncodingType.SEPARATED)) ? keyValueEncodingType : KeyValueEncodingType.INLINE;
properties.put("kv.encoding.type", String.valueOf(this.keyValueEncodingType));
```

Reply:
@tuteng 

No. I mean never allow passing in a `null` value.

so it would be:

```
checkNotNull(keyValueEncodingType, "Null encoding type is provided");
this.keyValueEncodingType = keyValueEncodingType;
properties.put("kv.encoding.type", String.valueOf(keyValueEncodingType));
```